pipelines/B2/B2.R

```diff
@@ -0,0 +1,71 @@
+# -------------------------------------------------------------------------
+# Step: Get the data
+# input: data/const/statbot_input_data.csv
+# output: ds as dataset class
+# -------------------------------------------------------------------------
+
+ds <- statbotData::create_dataset("B2")
+ds <- statbotData::download_data(ds)
+ds_boys <- ds$data %>%
+  janitor::clean_names() %>%
+  dplyr::mutate(gender = "boy")
+ds_boys
+# get additional datasets for girls
+ds$px_id <- "px-x-0104050000_102"
+ds$download_url <- "https://www.pxweb.bfs.admin.ch/DownloadFile.aspx?file=px-x-0104050000_102"
+ds <- statbotData::download_data(ds)
+
+ds_girls <- ds$data %>%
+  janitor::clean_names() %>%
+  dplyr::mutate(gender = "girl")
+ds_girls
+
+ds$data <- ds_boys %>%
+  dplyr::bind_rows(ds_girls)
+ds$data
+
+# -------------------------------------------------------------------------
+# Step: Clean the data
+#   input: ds$data
+#.  output: ds$cleaned_data: separate data by units
+# -------------------------------------------------------------------------
+
+ds$cleaned_data <- ds$data %>%
+  tidyr::pivot_wider(
+    names_from = unit_of_measure,
+    values_from = data_person
+  ) %>%
+  janitor::clean_names(
+  ) %>%
+  dplyr::rename(
+    amount = number,
+    canton = language_region_canton
+  ) %>%
+  dplyr::select(
+    c(gender, year, canton, first_name, year, amount, rank)
+  ) %>%
+  dplyr::filter(
+    !stringr::str_detect(canton, "speaking")
+  )
+ds$cleaned_data
```

Comment:
**issue:** at this stage, 91.6% of rank values for boys are 0 on my side:

```r
> ds$cleaned_data %>% dplyr::filter(gender=="boy")
# A tibble: 1,671,111 × 6
   gender year  canton      first_name amount  rank
   <chr>  <fct> <fct>       <fct>       <dbl> <dbl>
 1 boy    2000  Switzerland Noah          381     0
 2 boy    2001  Switzerland Noah          392     0
 3 boy    2002  Switzerland Noah          443     0
 4 boy    2003  Switzerland Noah          421     0
 5 boy    2004  Switzerland Noah          435     0
 6 boy    2005  Switzerland Noah          443     0
 7 boy    2006  Switzerland Noah          409     0
 8 boy    2007  Switzerland Noah          397     0
 9 boy    2008  Switzerland Noah          406     0
10 boy    2009  Switzerland Noah          406     0
```
```r
> boy_ranks <- ds$cleaned_data %>% dplyr::filter(gender=="boy") %>% pull(rank)
> sum(boy_ranks == 0) / length(boy_ranks)
[1] 0.9159027
```

Reply:
I ran the pipeline and got: 

```
> ds$postgres_export
# A tibble: 3,366,441 × 6
   gender year  first_name amount  rank spatialunit_uid
   <chr>  <fct> <fct>       <dbl> <dbl> <chr>          
 1 boy    2000  Noah          381     6 0_CH           
 2 boy    2001  Noah          392     4 0_CH           
 3 boy    2002  Noah          443     2 0_CH           
 4 boy    2003  Noah          421     2 0_CH           
 5 boy    2004  Noah          435     2 0_CH           
 6 boy    2005  Noah          443     2 0_CH           
 7 boy    2006  Noah          409     2 0_CH           
 8 boy    2007  Noah          397     2 0_CH           
 9 boy    2008  Noah          406     2 0_CH           
10 boy    2009  Noah          406     2 0_CH      
```